### ISSUE TITLE
Use system spellcheck

### ISSUE THREAD

**Issue by chchen2021** [Zettlr/Zettlr#2489]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [X] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [X] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

It will be really helpful to use the (macOS/Windows) system spellcheck instead of Zettlr's own custom spellcheck. 

This is a recap of the discussions in [#303](https://github.com/Zettlr/Zettlr/issues/303). I write bilingually in English and Chinese. Since there is no Chinese dictionary available, the English spellcheck will mark all Chinese characters as misspelled. It is very annoying and I find myself constantly switching the spellcheck on and off. 

### Example Workflow

Once Zettlr adopts system spellcheck, it should work smoothly without any user intervention. 

### Caveats

We need to inform users who have adopted Zettlr's current spellchecking system, or consider offering an option to keep their existing choice. 

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [ ] Windows
- [ ] macOS
- [ ] Linux

**Comment by nathanlesage** [comment]
Thanks for opening!

Regarding this:

> We need to inform users who have adopted Zettlr's current spellchecking system

I thought about replacing the custom spellcheck option with a message "Zettlr uses your OS's spellcheck settings"

**Comment by enricllagostera** [comment]
Could this be a way around for the Portuguese dictionary crash issue as well?

**Comment by nathanlesage** [comment]
> Could this be a way around for the Portuguese dictionary crash issue as well?

Yes!

**Comment by enricllagostera** [comment]
Great! That would be a big help for my use cases. I also write bilingually and have really missed having spellcheck support for PT-BR.

**Comment by vani-lu** [comment]
Another workaround might be allowing us to ignore certain characters with regex. Using the system dictionary can also be buggy for Chinese, as is my experience with the spellright extension in VSCode😿.

**Comment by nathanlesage** [comment]
Since this issue popped up in my notes again: I am actually wanting to switch to Electron's built-in spellchecker, but can't do so right now, since it doesn't offer an API access. They will have to implement this first, before we can follow suit.

**Comment by nathanlesage** [comment]
Due to recent upgrades, ALL hunspell-compatible spellcheck dictionaries should work as expected now. Given that there is still no API available from Electron, and it is not likely to come, I will consider this a `wontfix` for the time being.
